example-app/src/main/java/com/adyen/checkout/example/ui/compose/GenericDialog.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 26/8/2024.
+ */
+
+package com.adyen.checkout.example.ui.compose
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Button
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.adyen.checkout.example.ui.theme.ExampleTheme
+
+@Suppress("LongParameterList")
+@Composable
+fun GenericDialog(
+    modifier: Modifier = Modifier,
+    title: (@Composable () -> Unit)?,
+    content: (@Composable () -> Unit)?,
+    dismissButton: (@Composable () -> Unit)?,
+    confirmButton: (@Composable () -> Unit)?,
+    onDismiss: () -> Unit,
+) {
+    Dialog(onDismiss) {
+        Surface(
+            modifier = modifier.fillMaxWidth(),
+            shape = MaterialTheme.shapes.medium,
+            shadowElevation = 1.dp,
+        ) {
+            Column {
+                Column(Modifier.padding(ExampleTheme.dimensions.grid_2)) {
```

Comment:
I would avoid adding paddings in `GenericDialog`, since it does not create content, but gets it through arguments and each composables passed to the `GenericDialog` can define their own paddings.

It will also solve this pressed state background color not being fully visible.
![Screenshot 2024-09-11 at 14 31 29 (2)](https://github.com/user-attachments/assets/60729bd1-4e9d-4b0c-9554-819bf7cf7d4b)